src/lib.rs

```diff
@@ -242,4 +245,193 @@ mod test {
+                Txid::from_str("7d55e993274ba8d782a6b29f23e0517ac6ad7588252db8f1966c942d29ec055a")
+                    .unwrap(),
+                Txid::from_str("d04615a10516b1a108837724d24c69cce204ecfcde803f3e76b3062895ac0d6c")
+                    .unwrap(),
+            ],
+            pos: 2,
+        });
+        let merkle_proof = blocking_client.get_merkle_proof(&txid).unwrap();
+        let merkle_proof_async = async_client.get_merkle_proof(&txid).await.unwrap();
+        assert_eq!(merkle_proof, expected_merkle_proof);
+        assert_eq!(merkle_proof, merkle_proof_async);
+
+        // Test `get_output_status`
+        let expected_output_status = Some(OutputStatus {
+            spent: true,
+            txid: Some(
+                Txid::from_str("a23d84d15c30045cb515b7d1d93080d77cac895f9cb8d91e04ea4929714af575")
+                    .unwrap(),
+            ),
+            vin: Some(0),
+            status: Some(TxStatus {
+                confirmed: true,
+                block_height: Some(748689),
+                block_hash: Some(
+                    BlockHash::from_str(
+                        "000000000000000000053f3b9d2d44ae888ee84f0f989a8d15dba18e4d569ff8",
+                    )
+                    .unwrap(),
+                ),
+                block_time: Some(1660043384),
+            }),
+        });
+        let output_status = blocking_client.get_output_status(&txid, 1).unwrap();
+        let output_status_async = async_client.get_output_status(&txid, 1).await.unwrap();
+        assert_eq!(output_status, expected_output_status);
+        assert_eq!(output_status, output_status_async);
+
+        // Test `get_height`
+        let block_height = blocking_client.get_height().unwrap();
+        let block_height_async = async_client.get_height().await.unwrap();
+        assert_eq!(block_height, block_height_async);
+
+        // Test `get_tip_hash`
+        let tip_hash = blocking_client.get_tip_hash().unwrap();
+        let tip_hash_async = async_client.get_tip_hash().await.unwrap();
+        assert_eq!(tip_hash, tip_hash_async);
+
+        // Test `get_txid_at_block_index`
+        let expected_txid_at_block_index = Some(
+            Txid::from_str("5a278ad787373081130f534c99e7e795a1919ad20b9f8565ce70065c780b02c1")
+                .unwrap(),
+        );
+        let block_hash =
+            BlockHash::from_str("00000000000000000005a9417ce1cda81fcdcb94bc3e78efbdd121181ff01548")
+                .unwrap();
+        let txid_at_block_index = blocking_client
+            .get_txid_at_block_index(&block_hash, 2)
+            .unwrap();
+        let txid_at_block_index_async = async_client
+            .get_txid_at_block_index(&block_hash, 2)
+            .await
+            .unwrap();
+        assert_eq!(txid_at_block_index, expected_txid_at_block_index);
+        assert_eq!(txid_at_block_index, txid_at_block_index_async);
+
+        // Test `get_scripthash_txs`
+        let script = Script::new();
```

Comment:
While the empty script works for testing purposes here, this is quite slow since the calls return a lot of data. Suggestions for a really rare script are welcome to speed this up.